barretenberg/cpp/src/barretenberg/vm2/constraining/relations/alu.test.cpp

```diff
@@ -575,6 +575,244 @@ TEST(AluConstrainingTest, NegativeLTWrongTagCMismatch)
     EXPECT_THROW_WITH_MESSAGE(check_relation<alu>(trace), "C_TAG_CHECK");
 }
 
+// EQ TESTS
+
+auto process_eq_trace(const MemoryValue& test_a, const MemoryValue& test_b, bool equality = true)
+{
+    PrecomputedTraceBuilder precomputed_builder;
+
+    TestTraceContainer trace;
+    AluTraceBuilder builder;
+
+    auto mem_c = MemoryValue::from<uint1_t>(equality ? 1 : 0);
+
+    builder.process(
+        {
+            { .operation = simulation::AluOperation::EQ, .a = test_a, .b = test_b, .c = mem_c },
+        },
+        trace);
+
+    precomputed_builder.process_misc(trace, static_cast<uint8_t>(test_a.get_tag()) + 1);
+    precomputed_builder.process_tag_parameters(trace);
+    return trace;
+}
+
+auto process_eq_trace_different_tags(const MemoryValue& test_a, const MemoryValue& test_b, bool equality = true)
+{
+    PrecomputedTraceBuilder precomputed_builder;
+
+    TestTraceContainer trace;
+    AluTraceBuilder builder;
+
+    auto mem_c = MemoryValue::from<uint1_t>(equality ? 1 : 0);
+
+    builder.process(
+        {
+            { .operation = simulation::AluOperation::EQ,
+              .a = test_a,
+              .b = test_b,
+              .c = mem_c,
+              .error = simulation::AluError::TAG_ERROR },
+        },
+        trace);
+
+    precomputed_builder.process_misc(
+        trace, std::max(static_cast<uint8_t>(test_a.get_tag()), static_cast<uint8_t>(test_b.get_tag())) + 1);
```

Comment:
I guess the +1 is here in case we overflow the `u8`, which makes sense, but in the case we somehow have a tag of 255, then `precomputed_builder.process_misc(trace, 255)` would still fail because it won't generate enough rows (we start at 0)

Reply:
Ah I misread this - it's the max of the `a` and `b` tags which may differ. That makes sense since we still want the circuit to run in the case of a tag mismatch. I didn't use tag::max previously as I saw in another test we limited the number of rows to set up with the precomputed builder:
```
    // Build just enough clk rows for the lookup
```
If this doesn't really matter happy to move to tag::max!